src/plugins/unified_histogram/public/container/container.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { forwardRef, useEffect, useImperativeHandle, useMemo, useState } from 'react';
+import { Subject } from 'rxjs';
+import { UnifiedHistogramLayout, UnifiedHistogramLayoutProps } from '../layout';
+import type { UnifiedHistogramInputMessage } from '../types';
+import {
+  UnifiedHistogramState,
+  UnifiedHistogramStateOptions,
+  UnifiedHistogramStateService,
+} from './services/state_service';
+import { useStateProps } from './hooks/use_state_props';
+
+type LayoutProps = Pick<
+  UnifiedHistogramLayoutProps,
+  'services' | 'disableAutoFetching' | 'disableTriggers' | 'disabledActions'
+>;
+
+/**
+ * The props exposed by the container
+ */
+export type UnifiedHistogramContainerProps = Pick<
+  UnifiedHistogramLayoutProps,
+  'className' | 'resizeRef' | 'appendHitsCounter' | 'children'
+>;
+
+/**
+ * The options used to initialize the container
+ */
+export type UnifiedHistogramInitializeOptions = UnifiedHistogramStateOptions &
+  Omit<LayoutProps, 'services'>;
+
+/**
+ * The uninitialized API exposed by the container
+ */
+export interface UnifiedHistogramUninitializedApi {
+  /**
+   * Whether the container has been initialized
+   */
+  initialized: false;
+  /**
+   * Initialize the container
+   */
+  initialize: (options: UnifiedHistogramInitializeOptions) => void;
+}
+
+/**
+ * The initialized API exposed by the container
+ */
+export interface UnifiedHistogramInitializedApi {
+  /**
+   * Whether the container has been initialized
+   */
+  initialized: true;
```

Comment:
Its would be nice if we could find some stability in the react lifecycle to avoid needing this var. That said, I haven't worked on a stateful component like this before so it might be more difficult than I previously realized.

Reply:
This is primarily used to provide type safety to the consumer so they can't call API methods on an uninitialized API. Checking the `initialized` property on `UnifiedHistogramApi` will narrow it to either `UnifiedHistogramUninitializedApi` or `UnifiedHistogramInitializedApi` and allow calling the intended methods based on the initialization state.